improver/lapse_rate.py

```diff
@@ -290,80 +237,50 @@ def __repr__(self):
+        """Uses neighbourhood tools to pad and generate rolling windows
+        of the temp and orog datasets.
 
         Args:
-            temperature (1D numpy.ndarray):
-                Contains the temperature values for the central point and its
-                neighbours.
-
-            orography (1D numpy.ndarray):
-                Contains the height values for the central point and its
-                neighbours.
+            temp:
+                The temperature dataset.
+            orog:
+                The orography dataset.
 
         Returns:
-            float:
-                The gradient of the temperature/orography values. This
-                represents the lapse rate.
+            2 ndarrays, each containing rolling windows of the padded
+            datasets, return order: temp, orog.
 
         """
+        window_shape = (self.nbhood_size, self.nbhood_size)
+        orog_windows = neighbourhood_tools.pad_and_roll(
+            orog, window_shape, mode="constant", constant_values=np.nan
+        )
+        temp_windows = neighbourhood_tools.pad_and_roll(
+            temp, window_shape, mode="constant", constant_values=np.nan
+        )
+        return temp_windows, orog_windows
 
-        # If central point NaN then return blank value.
-        if np.isnan(temperature[self.ind_central_point]):
-            return DALR
-
-        # Remove points where there are NaN temperature values from both arrays
-        # before calculation.
-        y_data = temperature[~np.isnan(temperature)]
-        x_data = orography[~np.isnan(temperature)]
-
-        # Return DALR if standard deviation of both datasets = 0 (where all
-        # points are the same value).
-        if np.isclose(np.std(x_data), 0.0) and np.isclose(np.std(y_data), 0.0):
-            return DALR
-
-        matrix = np.stack([x_data, np.ones(len(x_data))], axis=0).T
-        gradient, _ = lstsq(matrix, y_data, rcond=None)[0]
-
-        return gradient
-
-    def _create_heightdiff_mask(self, all_orog_subsections):
-        """
-        Function to create a mask for any neighbouring points where the height
-        difference from the central point is greater than max_height_diff.
-
-        Slice through the orography subsection array to remove central
-        points.
-        Extracts the height value of each central point and masks out the
-        neighbouring points where their height difference is greater than
-        the maximum.
+    def _create_height_diff_mask(self, orog_subsections):
+        """Create a mask for any neighbouring points where the
+        height difference from the central point is greater than
+        max_height_diff.
 
         Args:
-            all_orog_subsections(2D numpy.ndarray):
-               Each row contains the height values of each neighbourhood.
+            orog_subsections:
+                3D numpy array where the final 2 axes represent the
+                orography neighbourhood data.
```

Comment:
Again, check out the normal style, which would make it like this:

```suggestion
            orog_subsections (numpy.ndarray):
                A 3-D numpy array where the leading axis ....(something)
                and the final 2 axes represent the orography
                neighbourhood data.
```

Reply:
Done